devtools/create-gossipstore.c

```diff
@@ -31,23 +51,38 @@ int main(int argc, char *argv[])
 			 "Send output to this file instead of stdout");
 	opt_register_arg("--input|-i", opt_set_charp, NULL, &infile,
 			 "Read input from this file instead of stdin");
+	opt_register_arg("--scidfile", opt_set_charp, NULL, &scidfile,
+			 "Input for 'scid, satshis' csv");
+	opt_register_arg("--sat", opt_set_charp, NULL, &csat,
+			 "default satoshi value if --scidfile flag not present");
 	opt_register_noarg("--help|-h", opt_usage_and_exit,
-			   "default-satoshis\n"
 			   "Create gossip store, from be16 / input messages",
 			   "Print this message.");
 
 	opt_parse(&argc, argv, opt_log_stderr_exit);
-	if (argc != 2)
-		errx(1, "Need default-satoshi argument for channel amounts");
-	if (!parse_amount_sat(&sat, argv[1], strlen(argv[1])))
-		errx(1, "Invalid satoshi amount %s", argv[1]);
+
+
+        if (scidfile) {
+		scidfd = fopen(scidfile, "r");
+		if (scidfd < 0)
+			err(1, "opening %s", scidfile);
+                scids = load_scid_file(scidfd);
+	        fclose(scidfd);
+	}
+	else if (csat) {
+		if (!parse_amount_sat(&sat, csat, strlen(csat))) {
+		        errx(1, "Invalid satoshi amount %s", csat);
+		}
+	}
+	else {
+		err(1, "must contain either --sat xor --scidfile");
```

Comment:
But your check doesn't validate this is true.  You probably want to check in the other branches that only one is set.